### PR TITLE
feat: support nullable api response data

### PR DESCRIPTION
**Pull request by everpcpc**

### CODE REVIEW

**Comment by everpcpc** [comment]
> need new test case to cover the change between 'NULL' and null

Yes, this PR should wait for changes in databend.

File: core/tests/core/simple.rs [hunk 1]
```diff
@@ -21,5 +21,14 @@ async fn select_simple() {
     let dsn = option_env!("TEST_DATABEND_DSN").unwrap_or(DEFAULT_DSN);
     let client = APIClient::new(dsn, None).await.unwrap();
     let resp = client.start_query("select 15532").await.unwrap();
-    assert_eq!(resp.data, [["15532"]]);
+    let result = resp
```
Comment (youngsofun) [inline core/tests/core/simple.rs:24]: assert_eq!(result, [[Some("15532")]]);

File: core/tests/core/simple.rs [hunk 2]
```diff
@@ -21,5 +21,14 @@ async fn select_simple() {
...
+        .data
+        .into_iter()
+        .map(|row| {
+            row.into_iter()
+                .map(|v| v.unwrap_or_default())
+                .collect::<Vec<_>>()
+        })
+        .collect::<Vec<_>>();
+    assert_eq!(result, [["15532"]]);
```
Comment (youngsofun) [inline core/tests/core/simple.rs:33]:
```suggestion
    assert_eq!(result, [[Some("15532")]]);
```
